processing/src/main/java/io/druid/query/FinalizeResultsQueryRunner.java

```diff
@@ -47,60 +43,17 @@ public FinalizeResultsQueryRunner(
   @Override
   public Sequence<T> run(final Query<T> query, Map<String, Object> responseContext)
   {
-    final boolean isBySegment = query.getContextBySegment(false);
     final boolean shouldFinalize = query.getContextFinalize(true);
-
     final Query<T> queryToRun;
-    final Function<T, T> finalizerFn;
-    final MetricManipulationFn metricManipulationFn;
-
     if (shouldFinalize) {
       queryToRun = query.withOverriddenContext(ImmutableMap.<String, Object>of("finalize", false));
-      metricManipulationFn = MetricManipulatorFns.finalizing();
-
     } else {
       queryToRun = query;
-      metricManipulationFn = MetricManipulatorFns.identity();
     }
-    if (isBySegment) {
-      finalizerFn = new Function<T, T>()
-      {
-        final Function<T, T> baseFinalizer = toolChest.makePostComputeManipulatorFn(
-            query,
-            metricManipulationFn
-        );
-
-        @Override
-        @SuppressWarnings("unchecked")
-        public T apply(T input)
-        {
-          Result<BySegmentResultValueClass<T>> result = (Result<BySegmentResultValueClass<T>>) input;
-
-          if (input == null) {
```

Comment:
@xvrl  pointed out this behavior is lost. I'll modify the PR to keep this behavior in general for null input rows